packages/flutter/test/scheduler/performance_mode_test.dart

```diff
@@ -0,0 +1,62 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
```

Comment:
I'd like to see the test framework updated to check if we have pending performance mode requests at the end of the test. That would help us make sure we're using correctly

Reply:
At the end of every test there is `expect(binding.getRequestedPerformanceMode(), isNull);` which is an expectation to ensure that there are no pending requests. Let me know if you think this is not sufficient.

Reply:
We have some generic post-test validation that runs and verifies that things like debugPlatformOverride is null, for example. I think that it could make sense to add there so that we validate it in every test scenario possilbe.

Unless we're not always finishing animations, which I guess is probably likely. Another pre/post step validation that is necessary is probably to reset the curent and requested performance mode so test state doesn't leak.

Reply:
Thanks for the pointer, I have added validation for performance mode disposal.